storm-client/src/jvm/org/apache/storm/grouping/LoadAwareShuffleGrouping.java

```diff
@@ -305,7 +309,7 @@ public static Scope downgrade(Scope current) {
             }
         }
 
-        public static Scope upgrade(Scope current) {
+        public static LocalityScope upgradeFromWorker(LocalityScope current) {
```

Comment:
This name can be a little confusing.  `upgradeFromWorker` implies the current scope is `WORKER_LOCAL`.

I would prefer to keep 'downgrade` and `upgrade`.